Assets resolving

mandryllo opened the pull request.

Commit:
Assets resolving

Commit:
Fix and cleanup

Commit:
Storage can be consumed as singleton

Commit:
Introduced default expiration time for s3 assets

underscope commented on server/content-repo/content-import.controller.js:
```diff
@@ -1,13 +1,11 @@
 const { ContentRepo } = require('../common/database');
-const config = require('../config');
-const createStorage = require('../common/storage');
+const Storage = require('../common/storage');
```
Watch the import order

underscope commented on server/content-repo/content-delivery.controller.js:
```diff
@@ -1,15 +1,14 @@
-const config = require('../config');
 const { createError } = require('../common/errors');
-const createStorage = require('../common/storage');
+const Storage = require('../common/storage');
```
Watch the import order

underscope commented on server/common/storage/index.js:
```diff
@@ -84,27 +108,7 @@ class Storage {
-}
-
-module.exports = createStorage;
+module.exports = new Storage(storage);
```
Storage service for repository content should be a separate module. One might use this adapter to create multiple services. Revert to previous state and implement content storage service as a separate module. Within it you can implement stuff you implemented as 'helpers' above.

Commit:
Revert "Storage can be consumed as singleton"

This reverts commit 7a96ae6fedef02d7e82cc21c52e0f3f64264f77e.

Commit:
Created Content Storage as separate module

Commit:
Fix getting wrong options in Storage :goat:

Commit:
Moved all content storage functions to content storage module

Commit:
Merge branch 'develop' into feature/assets-resolving

Commit:
Removed autobind & cleanup

underscope commented on server/content-repo/content-storage.js:
```diff
@@ -0,0 +1,88 @@
+
+const { storage } = require('../config');
+const Promise = require('bluebird');
+const createStorage = require('../common/storage');
```
Order 🤡

underscope commented on server/content-repo/content-storage.js:
```diff
@@ -0,0 +1,88 @@
+
+  _getRepoLocation(repoId, programId) {
+    return programId
+      ? `${this.options.importedContentLocation}/${programId}/${repoId}`
```
Since this is within ContentStorage service consider shortening location params to importPath and sourcePath.

underscope commented on server/content-repo/content-storage.js:
```diff
@@ -0,0 +1,88 @@
+    const _getUrl = key => {
+      return this.storage.getFileUrl(key)
+        .then(url => (asset.data.url = url))
+        .then(() => asset);
```
There is no need for this extra .then 💄

underscope commented on server/content-repo/content-storage.js:
```diff
@@ -0,0 +1,88 @@
+  }
+
+  _resolveAsset(asset) {
+    const _getUrl = key => {
```
Consider renaming this into processUrl (more descriptive). No need to prefix function with _ since it is wrapped within  _resolveAsset.

underscope commented on server/content-repo/content-storage.js:
```diff
@@ -0,0 +1,88 @@
+    return this.storage.getItem(key);
+  }
+
+  resolveContainer(container) {
```
What if somebody has array of elements that are not within container (i.e. assessment question). Consider renaming it to resolveElements and process array of elements.

underscope commented on server/content-repo/content-storage.js:
```diff
@@ -0,0 +1,88 @@
+
+  getAssessments(repoId, id, programId) {
+    const key =
+      `${this._getRepoLocation(repoId, programId)}/${id}.assessments.json`;
```
Consider resolving repoLocation on prev line.

Commit:
Cleanup

- fixed import order
- change resolveContainer to resolveElements
- change getUrl to processUrl
